app/api/auth/users/fetchUserData/nameTag/route.ts

```diff
@@ -0,0 +1,45 @@
+// Handle updating the user's nametag
+import { NameTagContent } from "@/components/NameTagForm";
+import startDB from "@/lib/db";
+import UserModel from "@/models/userModel";
+import { NextRequest, NextResponse } from "next/server";
+
+type FetchUserNametagResponse = NextResponse<{
+  success?: boolean;
+  error?: string;
+  nameTag?: NameTagContent;
+}>;
+
+// Make a GET request to fetch a user's nametag
+export const GET = async (req: NextRequest): Promise<FetchUserNametagResponse> => {
+    const userEmail = req.nextUrl.searchParams.get("userEmail");
+
+    if (userEmail) {
+      await startDB();
+
+      const user = await UserModel.findOne({ email: userEmail });
+
+      if (user) {
+        return NextResponse.json({
+          success: true,
+          nameTag: user?.nameTag
+        },
+        { status: 200 });
+      }
+      else {
+        return NextResponse.json({
+          success: false,
+          error: "User does not exist."
+        },
+        { status: 400 });
+      }
+    }
+
+    else {
+      return NextResponse.json({
+        success: false,
+        error: "User does not exist."
+      },
+      { status: 400 });
+    }
```

Comment:
If the two failure cases return the same thing, why not combine them?

```suggestion
      }
    }
    return NextResponse.json({
        success: false,
        error: "User does not exist."
      },
      { status: 400 });
```

Reply:
The second one is actually for if the "userEmail" param is not specified in the GET request. I updated the error message to reflect this: https://github.com/aimpowered/ZoomCompanion/pull/60/commits/4de12fc261848bfa27ddc6f420367d5dc239caf5